drivers/ma_yya_report/app/models/ma_yya_report/report.rb

```diff
@@ -0,0 +1,281 @@
+###
+# Copyright 2016 - 2022 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module MaYyaReport
+  class Report < SimpleReports::ReportInstance
+    include Rails.application.routes.url_helpers
+    include Reporting::Status
+
+    def run_and_save!
+      start
+      create_universe
+      report_results
+      complete
+    end
+
+    def start
+      update(started_at: Time.current)
+    end
+
+    def complete
+      update(completed_at: Time.current)
+    end
+
+    def title
+      'MA YYA Report'
+    end
+
+    def url
+      ma_yya_report_warehouse_reports_report_url(host: ENV.fetch('FQDN'), id: id, protocol: 'https')
+    end
+
+    private def create_universe
+      previous_period_filter = filter.deep_dup
+      previous_period_filter.end = filter.start - 1.day
+      previous_period_filter.start = previous_period_filter.start - 1.year
```

Comment:
This _should_ be roughly encapsulated in `filter.to_comparison`  Might need to set the `comparison_pattern` to get it right.

Reply:
maybe not though, depending on date ranges chosen, we might need a different comparison pattern?  Looks like you're getting the year prior to start, regardless of the size of the chosen range.